src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -293,24 +310,39 @@ object TaskGroupBuilder {
   }
 
   private[this] def computeContainerInfo(
-    podVolumes: Seq[raml.Volume],
+    podVolumes: Seq[Volume],
     container: MesosContainer): Option[mesos.ContainerInfo.Builder] = {
 
     val containerInfo = mesos.ContainerInfo.newBuilder.setType(mesos.ContainerInfo.Type.MESOS)
 
     container.volumeMounts.foreach { volumeMount =>
-      podVolumes.find(_.name == volumeMount.name).map { hostVolume =>
-        val volume = mesos.Volume.newBuilder
-          .setContainerPath(volumeMount.mountPath)
-
-        hostVolume.host.foreach(volume.setHostPath)
-
-        // Read-write mode will be used when the "readOnly" option isn't set.
-        val mode = if (volumeMount.readOnly.getOrElse(false)) mesos.Volume.Mode.RO else mesos.Volume.Mode.RW
-
-        volume.setMode(mode)
 
-        containerInfo.addVolumes(volume)
+      // Read-write mode will be used when the "readOnly" option isn't set.
+      val mode = if (volumeMount.readOnly.getOrElse(false)) mesos.Volume.Mode.RO else mesos.Volume.Mode.RW
+
+      podVolumes.find(_.name == volumeMount.name).foreach {
```

Comment:
`IllegalArgumentException` has special meaning from an API perspective (see MarathonExceptionMapper). Wouldn't a State exception be more clear as to the root of the problem (we've allowed the model to devolve into a bad state)?